packages/legend-query-builder/src/stores/QueryBuilderResultState.ts

```diff
@@ -113,12 +142,69 @@ export class QueryBuilderResultState {
     this.previewLimit = Math.max(1, val);
   };
 
+  addCellData = (val: CellData): void => {
+    this.selectedCells.push(val);
+  };
+
+  setSelectedCells = (val: CellData[]): void => {
+    this.selectedCells = val;
+  };
+
+  setMouseOverCell = (val: CellData | null): void => {
+    this.mousedOverCell = val;
+  };
+
   setQueryRunPromise = (
     promise: Promise<ExecutionResult> | undefined,
   ): void => {
     this.queryRunPromise = promise;
   };
 
+  findColumnFromCoordinates = (
+    colIndex: number,
+  ): string | number | boolean | null | undefined => {
+    if (
+      !this.executionResult ||
+      !(this.executionResult instanceof TDSExecutionResult)
+    ) {
+      return undefined;
+    }
+    return this.executionResult.result.columns[colIndex];
+  };
+
+  findRowFromRowIndex = (
+    rowIndex: number,
+  ): (string | number | boolean | null)[] => {
+    if (
+      !this.executionResult ||
+      !(this.executionResult instanceof TDSExecutionResult)
+    ) {
+      return [''];
+    }
+    return this.executionResult.result.rows[rowIndex]?.values ?? [''];
+  };
+
+  // NOTE: Due to the ways we edit Ag-grid's rendering, i.e.,
```

Comment:
Don't think we need this comment, it doesn't really clarify much here, potentially better to put it near where we use it